isthmus/src/main/java/io/substrait/isthmus/sql/SubstraitCreateStatementParser.java

```diff
@@ -0,0 +1,136 @@
+package io.substrait.isthmus.sql;
+
+import io.substrait.isthmus.SubstraitTypeSystem;
+import io.substrait.isthmus.calcite.SubstraitTable;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import org.apache.calcite.avatica.util.Casing;
+import org.apache.calcite.config.CalciteConnectionConfig;
+import org.apache.calcite.config.CalciteConnectionProperty;
+import org.apache.calcite.jdbc.CalciteSchema;
+import org.apache.calcite.jdbc.JavaTypeFactoryImpl;
+import org.apache.calcite.prepare.CalciteCatalogReader;
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.rel.type.RelDataTypeFactory;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.SqlNodeList;
+import org.apache.calcite.sql.ddl.SqlColumnDeclaration;
+import org.apache.calcite.sql.ddl.SqlCreateTable;
+import org.apache.calcite.sql.ddl.SqlKeyConstraint;
+import org.apache.calcite.sql.parser.SqlParseException;
+import org.apache.calcite.sql.parser.SqlParser;
+import org.apache.calcite.sql.parser.SqlParserPos;
+import org.apache.calcite.sql.parser.ddl.SqlDdlParserImpl;
+import org.apache.calcite.sql.validate.SqlConformanceEnum;
+import org.apache.calcite.sql.validate.SqlValidator;
+
+/** Utility class for parsing CREATE statements into a {@link CalciteSchema} */
+public class SubstraitCreateStatementParser {
+
+  private static final RelDataTypeFactory TYPE_FACTORY =
+      new JavaTypeFactoryImpl(SubstraitTypeSystem.TYPE_SYSTEM);
+
+  private static final CalciteConnectionConfig CONNECTION_CONFIG =
+      CalciteConnectionConfig.DEFAULT.set(
+          CalciteConnectionProperty.CASE_SENSITIVE, Boolean.FALSE.toString());
+
+  private static final SqlParser.Config PARSER_CONFIG =
+      SqlParser.config()
+          // To process CREATE statements we must use the SqlDdlParserImpl, as the default
+          // parser does not handle them
+          .withParserFactory(SqlDdlParserImpl.FACTORY)
+          .withUnquotedCasing(Casing.TO_UPPER)
+          .withConformance(SqlConformanceEnum.LENIENT);
+
+  private static final CalciteCatalogReader EMPTY_CATALOG =
+      new CalciteCatalogReader(
+          CalciteSchema.createRootSchema(false), List.of(), TYPE_FACTORY, CONNECTION_CONFIG);
+
+  // A validator is needed to convert the types in column declarations to Calcite types
+  private static final SqlValidator VALIDATOR =
+      new SubstraitSqlValidator(
+          // as we are validating CREATE statements, an empty catalog suffices
+          EMPTY_CATALOG);
+
+  /**
+   * @param createStatements a SQL string containing only CREATE statements
```

Comment:
for a valid javadoc it would make sense to also have a method description
```suggestion
   * Parses a SQL string containing only CREATE statements into a list of {@link SubstraitTable}s.
   *
   * @param createStatements a SQL string containing only CREATE statements
```